### DISCUSSION TITLE
Find port bound by `uv_udp_bind`

### DISCUSSION THREAD

**Discussion by smalls12** [libuv/libuv#3600] [Q&A]
I have the following code snippet:

```c
uv_udp_t send_socket_;
uv_udp_init(loop, &send_socket_);

struct sockaddr_in bind_addr;
uv_ip4_addr("0.0.0.0", 0, &bind_addr); // I believe this binds to all interfaces and random port ?

uv_udp_bind(&send_socket_, (const struct sockaddr *)&bind_addr, 0);
```

Just wondering if I can find the port that was randomly assigned.

Thank you.

**Comment by bnoordhuis** [comment]
You can retrieve it with [`uv_udp_getsockname()`](https://docs.libuv.org/en/v1.x/udp.html#c.uv_udp_getsockname).

**Reply by smalls12** [reply]
Thank you,

Just want to confirm I'm using this correctly.

```c
struct sockaddr_in addr_;
int len;
uv_udp_getsockname(&socket_, &addr_, &len);

char sender[17] = {0};
uv_ip4_name(&bind_addr, sender, len);

printf("Binded to %s:%d\n", sender, bind_addr.sin_port);
```

The address and ip are different than what tcpdump is reporting.

**Reply by bnoordhuis** [reply]
Check return values for errors. Also:
```c
struct sockaddr_in addr_;
int len = sizeof(addr_); // <- important!
uv_udp_getsockname(&socket_, &addr_, &len);
```
And:
```c
char sender[17] = {0};
uv_ip4_name(&bind_addr, sender, sizeof(sender));
```

**Reply by smalls12** [reply]
I implemented your changes and checked error codes but the IPs and Ports don't match tcpdump.
But I am convinced this is some NAT'ing related stuff on WSL.

But this isn't blocking me anyways.

Appreciate your time. :)
